File: Python/PROGRAMMERS/탐욕법/체육복.py
def solution(n, lost, reserve):
    arr = [1] * (n+1)
    for r in reserve:
        arr[r] +=1
    for l in lost:
        arr[l] -=1
    
    if arr[1] == 2 and arr[2] == 0:
        arr[1]=arr[2]=1
    if arr[n] == 2 and arr[n-1] ==0:
        arr[n]=arr[n-1]=1
    
    for i in range(2,n):
        if arr[i] == 2:

            if arr[i-1] == 0:
                arr[i-1] = 1
                arr[i] = 1
                continue

            if arr[i+1] == 0:
                arr[i+1] = 1
                arr[i] = 1
                continue
    answer=-1
    for a in arr:
        if a:
            answer+=1
    return answer

File: JavaScript/PROGRAMMERS/체육복.js
function solution(n, lost, reserve) {
    var answer = n;
    var students = []
    for(let i=0;i<n;i++) students.push(1)
    for(let i=0;i<reserve.length;i++) {
        var r = reserve[i]
        students[r-1] = 2
    }
    for(let i=0;i<lost.length;i++) {
        var l = lost[i]
        students[l-1] -= 1
        if (students[l-1] === 0) answer -= 1
    }
    for(let i=0;i<n;i++) {
        if (students[i] === 0) {
            if (0<i && i < n) {
                if (students[i-1] === 2) {
                    students[i] = 1
                    students[i-1] = 1
                    answer += 1
                    continue
                }
                if (students[i+1] === 2) {
                    students[i] = 1
                    students[i+1] = 1
                    answer += 1
                    
                }
            }
            if (i===0) {
                if (students[i+1] === 2) {
                    students[i+1] = 1
                    students[i] = 1
                    answer += 1
                }
            }
            if (i===n-1) {
                if (students[i-1] === 2) {
                    students[i-1] = 1
                    students[i] = 1
                    answer += 1
                }
            }
        }

    }
    console.log(answer,students)
    return answer;
}
solution(5,[2,3],[1,2])